libs/utils/decorate.js

```diff
@@ -1,29 +1,87 @@
+    if (buttonType === 'outline' && button.firstChild.nodeName === 'STRONG') buttonType = 'fill';
     button.classList.add('con-button', buttonType);
     if (size) button.classList.add(size); /* button-L, button-XL */
-    parent.insertAdjacentElement('afterend', button);
-    parent.remove();
+    const actionArea = button.closest('p');
+    if(actionArea) {
+      actionArea.classList.add('action-area');
+      if (idx === buttons.length - 1) {
+        actionArea.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
+      }
+    }
   });
-  const actionArea = buttons[0].closest('p');
-  actionArea.classList.add('action-area');
-  actionArea.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
 }
 
-export function decorateIcons(el) {
+export function decorateIconArea(el) {
   const icons = el.querySelectorAll('.icon');
   icons.forEach((icon) => {
     icon.parentElement.classList.add('icon-area');
     if (icon.textContent.includes('persona')) icon.parentElement.classList.add('persona-area');
   });
 }
 
+export async function getSVGsfromFile(path) {
+  if (!path) return null;
+  const resp = await fetch(path);
+  if (!resp.ok) return null;
+  const miloIcons = {};
+  const text = await resp.text();
+  const parser = new DOMParser();
+  const parsedText = parser.parseFromString(text, 'image/svg+xml');
+  const symbols = parsedText.querySelectorAll(`symbol`);
+  symbols.forEach( (symbol) => {
+    const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    while (symbol.firstChild) svg.appendChild(symbol.firstChild);
+    [...symbol.attributes].forEach((attr) => svg.attributes.setNamedItem(attr.cloneNode()));
+    svg.classList.add('icon-milo', `icon-milo-${svg.id}`);
+    miloIcons[svg.id] = svg;
+  });
+  return miloIcons;
+}
+
+// export async function decorateIconsInBlock(el) {
+//   const { miloLibs, codeRoot } = getConfig();
+//   const base = miloLibs || codeRoot;
+//   console.log(Object.keys(ICONS))
+//   if (Object.keys(ICONS).length === 0)
+//     await getSVGsfromFile(`${base}/img/icons/icons.svg`);
+//   if (!ICONS)
+//     return;
+//   const icons = el.querySelectorAll('span.icon');
+//   icons?.forEach(async (i) => {
+//     const iconName = i.classList[1].replace('icon-milo-', '');
+//     if (iconName === ICONS[iconName].id) {
+//       i.insertAdjacentHTML('afterbegin', ICONS[iconName].outerHTML);
+//     }
+//   });
+// }
+
+export async function loadIcons(el = document) {
+  const { miloLibs, codeRoot } = getConfig();
+  const base = miloLibs || codeRoot;
+  const miloIcons = await getSVGsfromFile(`${base}/img/icons/icons.svg`);
+  if (!miloIcons) return;
+  const icons = el.querySelectorAll('span.icon');
+  icons.forEach(async (i) => {
+    const iconName = i.classList[1].startsWith('icon-milo-') ? i.classList[1].replace('icon-milo-', '') : i.classList[1].replace('icon-', '');
+    if (iconName === miloIcons[iconName].id) {
```

Comment:
Isn't this always true for any valid icon id?

```suggestion
    if (!miloIcons[iconName]) return;
```

Reply:
Why yes, it is. Good call.